File: stock-mgmt-saas-multi-tenancy-ui/src/app/api-services/models/dashboard-stats-response.ts
import { MonthlyTrendEntry } from './monthly-trend-entry';

export interface DashboardStatsResponse {
  totalProducts?: number;
  totalCategories?: number;
  totalStockMvts?: number;
  lowStockAlerts?: number;
  newProductsThisMonth?: number;
  newCategoriesThisMonth?: number;
  newStockMvtsThisMonth?: number;
  monthlyTrend?: MonthlyTrendEntry[];
}

File: stock-mgmt-saas-multi-tenancy/src/main/java/com/tech/saas/responses/DashboardStatsResponse.java
package com.tech.saas.responses;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DashboardStatsResponse {

    private long totalProducts;
    private long totalCategories;
    private long totalStockMvts;
    private long lowStockAlerts;
    private long newProductsThisMonth;
    private long newCategoriesThisMonth;
    private long newStockMvtsThisMonth;
    private List<MonthlyTrendEntry> monthlyTrend;
}
